File: backend/router/mp_host.go
package router

import (
	v1 "github.com/1Panel-dev/1Panel/backend/app/api/v1"
	"github.com/1Panel-dev/1Panel/backend/middleware"

	"github.com/gin-gonic/gin"
)

type MpHostRouter struct{}

func (s *MpHostRouter) InitRouter(Router *gin.RouterGroup) {
	mpHostRouter := Router.Group("host").
		Use(middleware.JwtAuth()).
		Use(middleware.SessionAuth()).
		Use(middleware.PasswordExpired())
	baseApi := v1.ApiGroupApp.BaseApi
	{
		mpHostRouter.POST("create", baseApi.CreateMpHost)      // 新建mpHost表
		mpHostRouter.POST("delete", baseApi.DeleteMpHostByIds) // 批量删除mpHost表
		mpHostRouter.POST("update", baseApi.UpdateMpHost)      // 更新mpHost表
		mpHostRouter.POST("search", baseApi.GetMpHostList)
		mpHostRouter.GET("default/:id", baseApi.SetDefaultHost)
		mpHostRouter.GET("login/:id", baseApi.LoginMpHost)
		mpHostRouter.GET("refreshToken/:id", baseApi.LoginMpHost)
	}
}

File: frontend/src/api/mp/mp-host.ts
import http from '@/api/mp';
import { TimeoutEnum } from '@/enums/http-enum';
import { ResPage, SearchWithPage } from '@/api/interface';
import { MpHost } from '@/api/interface/mp-host';

export const searchMpHost = (params: SearchWithPage) => {
    return http.post<ResPage<MpHost.HostInfo>>(`/host/search`, params);
};
export const createMpHost = (params: MpHost.HostCreate) => {
    return http.post(`/host/create`, params, TimeoutEnum.T_40S);
};
export const updateMpHost = (params: MpHost.HostUpdate) => {
    return http.post(`/host/update`, params, TimeoutEnum.T_40S);
};
export const deleteMpHost = (params: MpHost.HostDelete) => {
    return http.post(`/host/delete`, params, TimeoutEnum.T_40S);
};
export const setDefaultHost = (params: { id: string }) => {
    return http.get(`/host/default/${params.id}`);
};
export const LoginMpHost = (params: { id: string }) => {
    return http.get(`/host/login/${params.id}`);
};
